tests/nest_tests/test_plasticity_dynamics.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# test_plasticity_dynamics.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import numpy as np
+import os
+import pytest
+
+import nest
+
+from pynestml.codegeneration.nest_tools import NESTTools
+from pynestml.frontend.pynestml_frontend import generate_nest_target
+
+try:
+    import matplotlib
+    matplotlib.use("Agg")
+    import matplotlib.ticker
+    import matplotlib.pyplot as plt
+    TEST_PLOTS = True
+except Exception:
+    TEST_PLOTS = False
+
+
+@pytest.fixture(autouse=True,
+                scope="module")
+def nestml_generate_target():
+    r"""Generate the neuron model code"""
+    files = [os.path.join("tests", "nest_tests", "resources", "test_plasticity_dynamics_neuron.nestml"),
+             os.path.join("tests", "nest_tests", "resources", "test_plasticity_dynamics_synapse.nestml")]
+    input_path = [os.path.realpath(os.path.join(os.path.dirname(__file__), os.path.join(
+        os.pardir, os.pardir, s))) for s in files]
+    generate_nest_target(input_path=input_path,
+                         logging_level="DEBUG",
+                         suffix="_nestml",
+                         codegen_opts={"neuron_synapse_pairs": [{"neuron": "test_plasticity_dynamics_neuron",
+                                                                 "synapse": "test_plasticity_dynamics_synapse",
+                                                                 "post_ports": ["post_spikes"]}],
+                                       "delay_variable": {"test_plasticity_dynamics_synapse": "d"},
+                                       "weight_variable": {"test_plasticity_dynamics_synapse": "w"}})
+
+
+@pytest.mark.skipif(NESTTools.detect_nest_version().startswith("v2"),
+                    reason="This test does not support NEST 2")
+def test_plasticity_dynamics():
+    r"""Test that the time interval-based plasticity rule in ``test_plasticity_dynamics_synapse`` is implemented correctly on a simple sequences of spikes"""
+    nest.ResetKernel()
+    nest.resolution = .01
+    nest.Install("nestmlmodule")
+
+    pre_spikes = nest.Create('spike_train_injector', params={'spike_times': [1, 10]})
+    post_driver_spikes = nest.Create('spike_train_injector', params={'spike_times': [.5]})
```

Comment:
I do not understand the logic behind your change in spike times. If I remember correctly, I carefully engineered the spike times in the two reproducer scripts to trigger the error.

Reply:
It was a more minimal set of spike times that reproduced the issue, but I have swapped it back out for your original times.